frontend/src/metabase-types/types/Metric.ts

```diff
@@ -1,6 +1,6 @@
 import { TableId } from "./Table";
 
-export type MetricId = number;
+export type MetricId = number | string;
```

Comment:
According to https://github.com/metabase/metabase/pull/21708/files#diff-235568e206ffee752d7db7d812a123b233a0c3bb7d1e749bfcef3bc87286ae12L47 there is a special case where the id can in fact be a string.